tools/link.py

```diff
@@ -3103,6 +3102,8 @@ def add_js_deps(sym):
           add_js_deps(sym)
         for sym in settings.EXPORTED_RUNTIME_METHODS:
           add_js_deps(shared.demangle_c_symbol_name(sym))
+        for sym in settings.EXPORTED_FUNCTIONS:
+          add_js_deps(shared.demangle_c_symbol_name(sym))
```

Comment:
Do the native methods not cause any problems here?

Reply:
No, `add_js_deps` here is really `maybe_add_js_deps` since it only does anyting if the symbol is actually found in the js_syms map.

Reply:
I see. I guess we never have overlap between the two sets of names since native ones are prefixed, so that seems safe for now. But if we remove the prefixing later maybe it would be an issue?

Reply:
It should still be fine.  This pass here is just adding any native deps for JS symbols we might want.  Its conservative, so the worst case would be that don't end up need the extra exports (i.e. if a native symbol satisfies the requirement the the JS symbol is not actually needed)  and metadce would remove them.

Reply:
Sounds good.